homeworks/vadym.myroshnyk_VadimMir/homework_5/homework_5.js

```diff
@@ -0,0 +1,81 @@
+// tesk 1
+const someArr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+function sumArrayFunction() {
+    let sum = 0;
+    return someArr.map((item) => {
+        sum = item + sum;
+        return sum;
+    });
+}
+
+console.log(sumArrayFunction());
+
+// tesk 2
+const classes = ['header', 'menu', 'menu_item', 'tabs',
+    'tab_item', 'menu', 'link', 'tabs', 'tab_item', 'menu',
+    'menu_item', 'menu', 'menu_item'];
+
+function returnObjKey() {
+    const result = {};
+
+    classes.reduce((previousValue, value, i) => {
+        let sumValue = 0;
+        sumValue = classes.filter(item => item === classes[i]).length;
+        result[value] = sumValue;
+        return result[value];
+    });
+    return result;
+}
+
+console.log(returnObjKey());
+
+// tesk 3
+const obj = {
+    b: 'c',
+    d: { e: 'f' },
+};
+
+function innerCopy() {
+    const newObj = {};
+    Object.keys(obj).forEach((value) => {
+        if (typeof (obj[value]) === 'object') {
+            for (let i = 0; i < obj[value]; i++) {
+                innerCopy(obj[value]);
+            }
+        }
+        newObj[value] = obj[value];
+    });
+    return newObj;
```

Comment:
this is also very hard to understand... `forEach` is enough, `for` loop is not needed

your task is as simple is `newObj[value] = typeof obj[value] === 'object' ? innerCopy(obj[value]) : obj[value]`;